pennylane/ops/identity.py

```diff
@@ -328,6 +328,29 @@ def compute_diagonalizing_gates(
         """
         return []
 
+    @staticmethod
+    def compute_decomposition(phi, wires=None):  # pylint:disable=arguments-differ,unused-argument
+        r"""Representation of the operator as a product of other operators (static method).
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        .. seealso:: :meth:`~.Identity.decomposition`.
+
+        Args:
+            phi (TensorLike): the global phase
+            wires (Any, Wires): A single wire that the operator acts on.
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> qml.GlobalPhase.compute_decomposition(wires=0)
+        []
+
+        """
+        return []
```

Comment:
Is there any way to tell if the `GlobalPhase` is being decomposed within the context of a controlled operation?

Reply:
not from within the function, but we could always add a check [here](https://github.com/PennyLaneAI/pennylane/blob/68b2f679d733ad31f1944cc0ae905993d54580a9/pennylane/ops/op_math/controlled.py#L596) and do something about it if we want to?

Reply:
Perhaps we can raise a warning? :thinking: I'm reluctant to raise needless warnings, but in this case we know the user will get (probably) wrong results.

Reply:
sure, I'll add the check + warning there.

Reply:
done!